presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -1063,13 +1062,13 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<Void> context)
                 return node;
             }
 
-            TableLayoutHandle newTableLayoutHandle = metadata.getAlternativeLayoutHandle(session, node.getLayout().get(), fragmentPartitioningHandle);
+            TableHandle newTableHandle = metadata.getAlternativeTableHandle(session, node.getTable(), fragmentPartitioningHandle);
             return new TableScanNode(
                     node.getId(),
-                    node.getTable(),
+                    newTableHandle,
                     node.getOutputSymbols(),
                     node.getAssignments(),
-                    Optional.of(newTableLayoutHandle),
+                    Optional.empty(),
```

Comment:
in theory this should be `newTableHandle.getLayout` . But if we decided to squash these commits it might not matter..